modules/proxy/mod_proxy_hcheck.c

```diff
@@ -680,7 +704,9 @@ static apr_status_t hc_check_cping(baton_t *baton, apr_thread_t *thread)
     set_request_connection(r, backend->connection);
     backend->connection->current_thread = thread;
 
-    if (hc->s->ping_timeout_set) {
+    if (hc->s->hc_timeout_set) {
+        timeout = hc->s->hc_timeout;
+    } else if (hc->s->ping_timeout_set) {
         timeout = hc->s->ping_timeout;
```

Comment:
I understand that the `hc_timeout` is the only configurable one specifically for the healthcheck mechanism, but I'm not sure that it should prevail on the (AJP) worker's `ping_timeout` if a AJP ping is configured (which is usually short anyway).
Maybe this change is not needed, or we should be able configure all the timeouts specifically for the healthcheck, but this could add more settings than necessary (i.e. never used, useless documentation..) so maybe we should wait for a real usecase?